packages/workspaces-extension/src/top_indicator.tsx

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) Jupyter Development Team.
+ * Distributed under the terms of the Modified BSD License.
+ */
+
+import {
+  blankIcon,
+  checkIcon,
+  LabIcon,
+  ReactWidget
+} from '@jupyterlab/ui-components';
+import React, { useEffect, useRef, useState } from 'react';
+import { searchIcon } from '@jupyterlab/ui-components';
+import { IWorkspacesModel } from '@jupyterlab/workspaces';
+
+interface IWorkspaceSelectorProps {
+  currentWorkspace: string;
+  identifiers: string[];
+  openWorkspace: (workspace: string) => void;
+  model?: IWorkspacesModel;
+}
+
+const WorkspaceSelector: React.FC<IWorkspaceSelectorProps> = ({
+  currentWorkspace,
+  identifiers,
+  openWorkspace
+}) => {
+  const [isExpanded, setIsExpanded] = useState(false);
+  const [searchQuery, setSearchQuery] = useState('');
+  const dropdownRef = useRef<HTMLDivElement>(null);
+
+  const filteredIdentifiers = identifiers.filter(identifier =>
+    identifier.toLowerCase().includes(searchQuery.toLowerCase())
+  );
+
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (
+        dropdownRef.current &&
+        !dropdownRef.current.contains(event.target as Node)
+      ) {
+        setIsExpanded(false);
+      }
+    };
+
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => document.removeEventListener('mousedown', handleClickOutside);
+  }, []);
+
+  return (
+    <div className="jp-WorkspaceSelector" ref={dropdownRef}>
+      <button
+        className="jp-WorkspaceSelectorHeader"
+        onClick={() => setIsExpanded(!isExpanded)}
+        aria-expanded={isExpanded}
+      >
+        <span className="jp-WorkspaceSelectorCurrent">
+          {currentWorkspace.length > 12
+            ? `${currentWorkspace.slice(0, 12)}...`
+            : currentWorkspace}
+        </span>
+        <span className="jp-WorkspaceSelectorCaret">
+          {isExpanded ? '▲' : '▼'}
+        </span>
+      </button>
+
+      {isExpanded && (
+        <div className="jp-WorkspaceSelectorDropdown">
+          <div className="jp-WorkspaceSelectorSearch">
+            <div className="jp-WorkspaceSelectorSearchIcon">
+              <LabIcon.resolveReact icon={searchIcon} />
+            </div>
+            <input
+              type="text"
+              className="jp-WorkspaceSelectorInput"
+              placeholder="Search"
+              value={searchQuery}
+              onChange={e => setSearchQuery(e.target.value)}
+              autoFocus
```

Comment:
(this is my best effort on getting the accessibility roles right based on https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-autocomplete-list/)

```suggestion
              autoFocus
              aria-autocomplete="list"
              role="combobox"
```